release.nix

```diff
@@ -8,18 +8,48 @@
   }
 }:
 
-with (import ./nix/release-lib.nix) {
+let
+  # `release-lib.nix` calls `packageSet` many times. This
+  # results in lots of repeated work, both during eval stage
+  # and when the jobs are run.
+  # By memoizing the calls here we get a very significant
+  # speedup of the eval time on hydra.
+  packageSet' = import cardano-wallet;
+  packageSetMemo = json:
+    { inherit json; packageSet = packageSet' (builtins.fromJSON json); };
+  packageSets =
+    [
+    # TODO : find a better way to populate this list
+    (packageSetMemo (''{"config":{"allowUnfree":false,"allowUnsupportedSystem":true,"inHydra":true},"gitrev":"'' + projectArgs.gitrev + ''","system":"x86_64-linux"}''))
+    (packageSetMemo (''{"config":{"allowUnfree":false,"inHydra":true},"crossSystem":{"config":"x86_64-pc-mingw32","libc":"msvcrt","platform":{}},"gitrev":"'' + projectArgs.gitrev + ''","system":"x86_64-linux"}''))
+    (packageSetMemo (''{"config":{"allowUnfree":false,"inHydra":true},"gitrev":"'' + projectArgs.gitrev + ''","system":"x86_64-darwin"}''))
+    (packageSetMemo (''{"config":{"allowUnfree":false,"inHydra":true},"gitrev":"'' + projectArgs.gitrev + ''","system":"x86_64-linux"}''))
+    (packageSetMemo (''{"config":{"allowUnfree":false,"inHydra":true},"crossSystem":{"config":"x86_64-unknown-linux-musl"},"gitrev":"'' + projectArgs.gitrev + ''","system":"x86_64-linux"}''))
+    ];
+in
+
+with (import ./nix/release-lib.nix) rec {
   inherit (import ./nix/iohk-common.nix {}) pkgs;
   inherit supportedSystems supportedCrossSystems scrubJobs projectArgs;
-  packageSet = import cardano-wallet;
+  packageSet = args:
+    (pkgs.lib.findFirst (x: x.json == builtins.toJSON args)
+      (builtins.trace ("Warning no memoization for " + builtins.toJSON args) { packageSet = packageSet' args; })
+        packageSets).packageSet;
   gitrev = cardano-wallet.rev;
 };
 
 with pkgs.lib;
 
 let
   testsSupportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
-  collectTests = ds: filter (d: elem d.system testsSupportedSystems) (collect isDerivation ds);
+  # Adds the package name to the test derivations for windows-testing-bundle.nix
+  # (passthru.identity.name does not seem to survive release-lib.nix)
```

Comment:
If passthru.identity.name gets removed, then I think it would be just be simplest to remove this change and the corresponding one in windows-testing-bundle.nix.

Reply:
The problem is that `$pkg` is no longer the packager name.  Tests and benchmarks are now in `bin` subdir (not a package id based subdir). I'll replace `$pkg` with `"bin"` to make this clear.

We need to ensure that the executable in the windows bundle has a unique location (otherwise tests like `unit-tests` will clash when we copy them to the bundle).  We could zip [1..] with the list and put each executable in a numbered sub dir, but I think it might be more user friendly to keep using the package name somehow.

Reply:
OK got it. I have split the function and added an extra comment.